src/app/CommandPathRegistry.h

```diff
@@ -0,0 +1,116 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <stddef.h>
+
+#include <app/ConcreteCommandPath.h>
+#include <lib/core/CHIPError.h>
+#include <lib/core/Optional.h>
+
+namespace chip {
+namespace app {
+
+struct CommandPathRegistryEntry
+{
+    ConcreteCommandPath requestPath = ConcreteCommandPath(0, 0, 0);
+    Optional<uint16_t> ref;
+};
+
+class CommandPathRegistry
+{
+public:
+    virtual ~CommandPathRegistry(){};
+    virtual Optional<CommandPathRegistryEntry> Find(const ConcreteCommandPath & requestPath) const  = 0;
+    virtual Optional<CommandPathRegistryEntry> GetFirstEntry() const                                = 0;
+    virtual CHIP_ERROR Add(const ConcreteCommandPath & requestPath, const Optional<uint16_t> & ref) = 0;
+    virtual size_t Count() const                                                                    = 0;
+    virtual size_t MaxSize() const                                                                  = 0;
+};
+
+/**
+ * @class BasicCommandPathRegistry
+ *
+ * @brief Allows looking up CommandRef using the requested ConcreteCommandPath.
+ *
+ * While there are faster implementations, right now batch commands are capped at a low number due to
+ * message size constraints. All commands need to be contained within a single InvokeRequest. In
+ * practice this is usually less than 60 commands (but could be much more with TCP transports or
+ * newer transports).
+ */
+template <size_t N>
+class BasicCommandPathRegistry : public CommandPathRegistry
+{
+public:
+    ~BasicCommandPathRegistry() override{};
```

Comment:
Please do not override to empty.

Reply:
Cone of shame..... Good catch. I threw this in there when I was getting weird compiler errors about not finding the vtable for the destructor, I was experimenting but forgot to remove this.

The real error was I forgot the `=0` for one of the interface to make it a pure virtual